diff/parse.go

```diff
@@ -72,6 +73,12 @@ func (r *MultiFileDiffReader) ReadFile() (*FileDiff, error) {
 		}
 	}
 
+	// FileDiff is added/deleted file
+	// No further collection of hunks needed
+	if fd.NewName == "" {
```

Comment:
```suggestion
	if fd.NewName == "" || fd.OldName == "" {
```

Doesn't this need to check for the old file name too to catch the case where we're creating/adding a file?

Reply:
All filenames from `Only in {source}: {filename}` messages will be added to `OrigName` field.